File: src/main/java/com/company/commitet_jm/entity/StatusSheduler.java
package com.company.commitet_jm.entity;

import io.jmix.core.metamodel.datatype.EnumClass;

public enum StatusSheduler implements EnumClass<String> {
    NEW("A"),
    PROCESSED("B"),
    COMPLETE("C"),
    ERROR("D");

    private final String id;

    StatusSheduler(String id) {
        this.id = id;
    }

    @Override
    public String getId() {
        return id;
    }

    public static StatusSheduler fromId(String id) {
        for (StatusSheduler status : StatusSheduler.values()) {
            if (status.id.equals(id)) {
                return status;
            }
        }
        return null;
    }
}

File: bin/main/com/company/commitet_jm/entity/StatusSheduler.kt
package com.company.commitet_jm.entity

import io.jmix.core.metamodel.datatype.EnumClass

enum class StatusSheduler(private val id: String) : EnumClass<String> {
    NEW("A"),
    PROCESSED("B"),
    COMPLETE("C"),
    ERROR("D");

    override fun getId() = id

    companion object {

        @JvmStatic
        fun fromId(id: String): StatusSheduler? = StatusSheduler.values().find { it.id == id }
    }
}